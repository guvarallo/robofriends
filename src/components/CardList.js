import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
  return (
    <>
      {robots.map((user, i) => {
        return (
          <Card
            key={robots[i].id}
            id={robots[i].id}
            firstName={robots[i].firstName}
            lastName={robots[i].lastName}
            email={robots[i].email}
          />
        );
      })}
    </>
  );
};

export default CardList;
