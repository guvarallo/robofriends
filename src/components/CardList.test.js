import React from 'react';
import renderer from 'react-test-renderer';

import CardList from './CardList';

it('expect to render CardList component', () => {
  const mockRobots = [
    {
      id: 1,
      firstName: 'John',
      lastName: 'Snow',
      email: 'john@test.com',
    },
  ];
  const tree = renderer.create(<CardList robots={mockRobots} />).toJSON();
  expect(tree).toMatchSnapshot();
});
