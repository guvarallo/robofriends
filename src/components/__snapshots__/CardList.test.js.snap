// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
<div
  className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="robots"
    src="https://robohash.org/1?size=200x200"
  />
  <h2>
    John
     
    Snow
  </h2>
  <p>
    john@test.com
  </p>
</div>
`;
